src/status_im/ui/screens/discover/subs.cljs

```diff
@@ -25,44 +40,53 @@
             (vals discoveries))))
 
 (reg-sub :get-popular-discoveries
-  (fn [db [_ limit tags]]
-    (let [discoveries (:discoveries db)
-          current-tag (:current-tag db)
-          search-tags (:discover-search-tags db)]
-      (let [discoveries (->> (get-discoveries-by-tags discoveries current-tag (or tags search-tags))
-                             (map #(assoc % :priority (calculate-priority db %)))
-                             (sort-by :priority >))]
-        {:discoveries (take limit discoveries)
-         :total       (count discoveries)}))))
+  :<- [:get-discoveries]
+  :<- [:get-current-tag]
+  :<- [:get-discover-search-tags]
+  :<- [:chats]
+  :<- [:get-contacts]
+  :<- [:get-current-public-key]
+  (fn [[discoveries current-tag discover-search-tags chats contacts public-key]
+       [_ limit tags]] 
+    (let [discoveries (->> (get-discoveries-by-tags discoveries
+                                                    current-tag
+                                                    (or tags discover-search-tags))
+                           (map #(assoc % :priority (calculate-priority chats public-key contacts %)))
+                           (sort-by :priority >))]
+      {:discoveries (take limit discoveries)
+       :total       (count discoveries)})))
 
 (reg-sub :get-top-discovery-per-tag
-  (fn [{:keys [discoveries tags]} [_ limit]]
+  :<- [:get-discoveries]
+  :<- [:get-tags]
+  (fn [[discoveries tags] [_ limit]]
     (let [tag-names (map :name (take limit tags))]
-     (for [tag tag-names]
-       (let [results (get-discoveries-by-tags discoveries tag nil)]
+      (for [tag tag-names]
+        (let [results (get-discoveries-by-tags discoveries tag nil)]
           [tag {:discovery (first results)
                 :total     (count results)}])))))
 
 (reg-sub :get-recent-discoveries
-  (fn [db]
-    (sort-by :created-at > (vals (:discoveries db)))))
+  :<- [:get-discoveries]
+  (fn [discoveries]
+    (sort-by :created-at > (vals discoveries))))
 
 (reg-sub :get-popular-tags
-  (fn [db [_ limit]]
-    (take limit (:tags db))))
+  :<- [:get-tags]
+  (fn [tags [_ limit]]
+    (take limit tags)))
 
 (reg-sub :get-discover-search-results
-  (fn [db]
-    (let [discoveries (:discoveries db)
-          current-tag (:current-tag db)
-          tags        (:discover-search-tags db)]
-      (get-discoveries-by-tags discoveries current-tag tags))))
+  :<- [:get-discoveries]
+  :<- [:get-current-tag]
+  :<- [:get-discover-search-tags]
+  (fn [[discoveries current-tag discover-search-tags]]
+    (get-discoveries-by-tags discoveries current-tag discover-search-tags)))
 
+;; TODO janherich: is this really what we want, what about `state-of-dapps` group ?
```

Comment:
Can we clarify this in the chat, and create an issue if needed?

Reply:
This is list of DApps on discover screen, nothing to do with chat

Reply:
I meant in slack :) To clarify if this is indeed what we really want.

Reply:
Intentional. It is a temporary list of dapps, until idea `#14` is implemented.